### PR TITLE
Remove deprecated syntax: '#' and 'kind'

### PR DESCRIPTION
**Pull request by JordanMartinez**
**Description of the change**

Breaking changes that will be made in `v0.15.x`. Removes syntax that was deprecated in `v0.14.x`.

`# Type` was replaced with `Row Type`:
```purescript
-- 0.13.x
data RProxy (rows :: # Type) = RProxy
-- 0.14.x
data RProxy (rows :: Row Type) = RProxy
```
kind declarations, imports, and exports no longer require use a keyword of 'kind'.

```purescript
-- 0.13.x
module Foo (kind MyKind) where

import Bar (kind AnotherKind)

foreign import kind MyKind
foreign import data Foo :: MyKind
```
```purescript
-- 0.14.x
module Foo (MyKind) where

import Bar (AnotherKind)

data MyKind
foreign import data Foo :: MyKind
```

---

**Checklist:**

- [x] Added a file to CHANGELOG.d for this PR (see CHANGELOG.d/README.md)
- [ ] Added myself to CONTRIBUTORS.md (if this is my first contribution)
- [ ] Linked any existing issues or proposals that this pull request should close
- [ ] Updated or added relevant documentation
- [ ] Added a test for the contribution (if applicable)

### CODE REVIEW

**Comment by JordanMartinez** [comment]
To clarify, should the warnings be updated to errors? Or are they removed completely as I've done in this PR?

File: lib/purescript-cst/src/Language/PureScript/CST/Parser.y [hunk 1]
```diff
@@ -304,16 +297,12 @@ type2 :: { Type () }
   | type3 '=>' type1 {% do cs <- toConstraint $1; pure $ TypeConstrained () cs $2 $3 }
 
 type3 :: { Type () }
-  : type4 { $1 }
+  : type4 %shift { $1 }
   | type3 qualOp type4 { TypeOp () $1 (getQualifiedOpName $2) $3 }
```
Comment (rhendric) [inline lib/purescript-cst/src/Language/PureScript/CST/Parser.y:301]: This'll fix your last conflict:

```suggestion
  | type3 qualOp type4 %shift { TypeOp () $1 (getQualifiedOpName $2) $3 }
```

Comment (JordanMartinez) [reply lib/purescript-cst/src/Language/PureScript/CST/Parser.y]: Thanks!
How were you able to figure that out?

Comment (rhendric) [reply lib/purescript-cst/src/Language/PureScript/CST/Parser.y]: In this case, the `type4 : type5 %shift` production told Happy to disregard this reduction if anything under `type5` can keep being greedy. Removing that production meant that productions that end in `type4` would themselves need to be given `%shift` precedence, or else the shift-reduce conflicts being prevented by the `%shift` in `type4 : type5 %shift` would re-emerge (which is, I assume, what happened here).

Comment (JordanMartinez) [reply lib/purescript-cst/src/Language/PureScript/CST/Parser.y]: Ah! I see! That makes sense. Thanks for explaining that. :smiley:

**Review by rhendric — APPROVED** [review]

**Pull request merged**